cuda_core/tests/test_program.py

```diff
@@ -150,3 +227,105 @@ def test_program_close():
     program = Program(code, "c++")
     program.close()
     assert program.handle is None
+
+
+
+
+def test_nvvm_deferred_import():
+    """Test that our deferred NVVM import works correctly"""
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+        nvvm = _get_nvvm_module()
+        assert nvvm is not None
+    except ImportError as e:
+        pytest.skip(f"NVVM not available: {e}")
+
+
+def test_nvvm_program_creation():
```

Comment:
ditto, fixtures can be passed to tests this way (pick a name and stick to it)
```suggestion
def test_nvvm_program_creation(nvvm_ir):
```

Reply:
The skip condition should be used to decorate all NVVM-related tests, so that we can remove try-except in the test bodies.